src/mono/wasm/runtime/binding_support.js

```diff
@@ -2108,19 +2332,236 @@ var BindingSupportLib = {
 			if (!BINDING._use_finalization_registry) {
 				listener[BINDING.listener_registration_count_symbol]--;
 				if (listener[BINDING.listener_registration_count_symbol] === 0) {
-					BINDING._js_owned_object_table.delete(listener_gc_handle);
-					BINDING._release_js_owned_object_by_gc_handle(listener_gc_handle);
+					BINDING._js_owned_object_finalized(listener_gc_handle);
 				}
 			}
 
 			return 0;
 		} catch (exc) {
-			return BINDING.js_string_to_mono_string(exc.message);
+			return BINDING.wrap_error(null, exc.message);
 		} finally {
 			nameRoot.release();
 		}
 	},
+	mono_wasm_cancel_promise: function (thenable_js_handle, is_exception) {
+		try {
+			const promise = BINDING.mono_wasm_get_jsobj_from_js_handle(thenable_js_handle)
+			const promise_control = promise[BINDING.promise_control_symbol];
```

Comment:
mono_wasm_get_jsobj_from_js_handle can return null, we should handle that case here

Reply:
I was more worried about race condition with CancellationTokenRegistration. I added condition to make sure that only incomplete tasks are canceled.